storage/badger/procedure/executed.go

```diff
@@ -11,28 +11,10 @@ import (
 	"github.com/onflow/flow-go/storage/badger/operation"
 )
 
-// UpdateHighestExecutedBlockIfHigher updates the latest executed block to be the input block
-// if the input block has a greater height than the currently stored latest executed block.
-// The executed block index must have been initialized before calling this function.
-// Returns storage.ErrNotFound if the input block does not exist in storage.
-func UpdateHighestExecutedBlockIfHigher(header *flow.Header) func(txn *badger.Txn) error {
+// UpdateLastExecutedBlock updates the latest executed block to be the input block
+func UpdateLastExecutedBlock(header *flow.Header) func(txn *badger.Txn) error {
 	return func(txn *badger.Txn) error {
-		var blockID flow.Identifier
-		err := operation.RetrieveExecutedBlock(&blockID)(txn)
-		if err != nil {
-			return fmt.Errorf("cannot lookup executed block: %w", err)
-		}
-
-		var highest flow.Header
-		err = operation.RetrieveHeader(blockID, &highest)(txn)
-		if err != nil {
-			return fmt.Errorf("cannot retrieve executed header: %w", err)
-		}
-
-		if header.Height <= highest.Height {
-			return nil
-		}
```

Comment:
This is the logic being simplified

Reply:
Not sure I understand why we can simplify this logic away. Can you give a short explanation why we don't need this anymore?

Reply:
[I explained in this issue](https://github.com/onflow/flow-go/issues/6908#issue-2793424515)